### PR TITLE
fix: block `distribution` module account to receive ZETA

### PR DESCRIPTION
**Pull request by brewmaster012**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [x] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -196,6 +196,11 @@ var (
 		emissionsModuleTypes.UndistributedObserverRewardsPool: nil,
 		emissionsModuleTypes.UndistributedTssRewardsPool:      nil,
 	}
+disa
+	// module accounts that are allowed to receive tokens
```
Comment (kingpinXD) [inline app/app.go:200]: NIT :  module accounts that are not allowed to receive tokens

File: app/app.go [hunk 2]
```diff
@@ -196,6 +196,11 @@ var (
...
+	blockedReceivingModAcc = map[string]bool{
```
Comment (kingpinXD) [inline app/app.go:201]: Might be a good idea to iterate over all module accounts and set them all to false by default , and we can turn on for the ones that w need to allow to receive funds 
`app.ModuleAccountAddrs()`

**Review by lumtis — APPROVED** [review]
utACK

File: app/app.go (hunk 1 above)
Comment (brewmaster012) [reply app/app.go:200]: done

**Review by kingpinXD — APPROVED** [review]

**Pull request merged**